packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -500,6 +500,23 @@ export class TaskDefinition extends TaskDefinitionBase {
 
     return this.containers.map(x => x.renderContainerDefinition());
   }
+
+  private renderVolumes(): CfnTaskDefinition.VolumeProperty[] {
+    return this.volumes.map(renderVolume);
+
+    function renderVolume(spec: Volume): CfnTaskDefinition.VolumeProperty {
+      return {
+        ...spec,
+        efsVolumeConfiguration: spec.efsVolumeConfiguration && {
+          authorizationConfig: spec.efsVolumeConfiguration.authorizationConfig,
+          filesystemId: spec.efsVolumeConfiguration.fileSystemId,
+          rootDirectory: spec.efsVolumeConfiguration.rootDirectory,
+          transitEncryption: spec.efsVolumeConfiguration.transitEncryption,
+          transitEncryptionPort: spec.efsVolumeConfiguration.transitEncryptionPort,
+        },
+      };
+    }
+  }
```

Comment:
This whole madness is 100% caused by the case changing from `FileSystemId` to `FilesystemId`.